src/main/java/graphql/nadel/engine/ServiceResultNodesToOverallResult.java

```diff
@@ -67,8 +67,17 @@ private ListExecutionResultNode mapListExecutionResultNode(ListExecutionResultNo
     }
 
     private ObjectExecutionResultNode mapObjectResultNode(ObjectExecutionResultNode objectResultNode, GraphQLSchema overallSchema, Map<Field, FieldTransformation> transformationMap) {
-        FetchedValueAnalysis fetchedValueAnalysis = fetchedAnalysisMapper.mapFetchedValueAnalysis(objectResultNode.getFetchedValueAnalysis(), overallSchema, transformationMap);
-        return new ObjectExecutionResultNode(fetchedValueAnalysis, objectResultNode.getChildren());
+        FetchedValueAnalysis originalFetchAnalysis = objectResultNode.getFetchedValueAnalysis();
+        MergedField originalField = originalFetchAnalysis.getExecutionStepInfo().getField();
+        FetchedValueAnalysis fetchedValueAnalysis = fetchedAnalysisMapper.mapFetchedValueAnalysis(originalFetchAnalysis, overallSchema, transformationMap);
+
+        objectResultNode = new ObjectExecutionResultNode(fetchedValueAnalysis, objectResultNode.getChildren());
+
+        FieldTransformation fieldTransformation = transformationMap.get(originalField.getSingleField());
+        if (fieldTransformation != null) {
+            objectResultNode = fieldTransformation.unapplyResultNode(objectResultNode);
+        }
+        return objectResultNode;
     }
```

Comment:
This is a 2 step mapping - first the FetchedValueAnalysis is changed (say with field renames and __typename being removed) and then the actual result objects are changed (eg remove the __typename) in the response